src/components/widgets/search-widget/search-widget.js

```diff
@@ -33,9 +32,13 @@ const SearchWidget = ({ view }) => {
     setSearchWidget(null);
   }
   
+  async function addSearchWidgetToView() {
+    await view.ui.add(searchWidget, "top-left");
+  }
+  
   useEffect(() => {
     if( searchWidget ) {
-      view.ui.add(searchWidget, "top-left");
+      addSearchWidgetToView().then(() => document.querySelector(".esri-search__input").focus());
```

Comment:
alternatively, not to use `then` syntax you could do:
```
const addSearchWidgetToView = async () => {
    await view.ui.add(searchWidget, "top-left");
    document.querySelector(".esri-search__input").focus();
};
addSearchWidgetToView();
```
inside a hook. You cannot wrap the whole useEffect with async function (`useEffect(async () => { ... })`) because `useEffect` has to return a clean up function but what you can do is define a named inner function (addSearchWidgetToView) that you will then call inside useEffect; it will be run asynchronously but the statements inside `addSearchWidgetToView` itself will be executed synchronously (so you don't have to worry that you will call `focus` on not yet existing node.